custom_components/tuya_local/devices/hotshot2000_heater.yaml

```diff
@@ -0,0 +1,72 @@
+name: HOT SHOT 2000 WIFI
+products:
+  - id: lgn0xaegjbodrjax
+    name: EUROM Hot-Shot 2000 WiFi
+primary_entity:
+  entity: climate
+  dps:
+    - id: 1
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          constraint: work_mode
+          conditions:
+            - dps_val: level1
+              value: fan_only
+            - dps_val: level2
+              value: heat
+            - dps_val: level3
+              value: heat
+    - id: 2
+      type: integer
+      name: temperature
+      range:
+        min: 5
+        max: 37
+    - id: 3
+      type: integer
+      name: current_temperature
+    - id: 5
+      name: work_mode
+      type: string
+      hidden: true
+    - id: 6
+      name: preset_mode
+      type: boolean
+      mapping:
+        - dps_val: false
+          constraint: work_mode
+          conditions:
```

Comment:
This is missing the "level1" condition.  If you don't want it showing up as an option, you can mark it hidden, but it would help to have a value mapped to it to show in the UI.

Reply:
As I understood from the docs for the Climate Entity, the "preset_mode" does not have "fan_only" option. The "level1" is present under the "hvac_mode" which supports "heat" and "fan_only", see line 17. And it is selectable in the UI:

![image](https://github.com/make-all/tuya-local/assets/5202136/e657e01e-3b17-4487-9546-e7680be45604)

Reply:
Unlike HVAC mode, preset mode does not have to be from the predefined modes, so it won't hurt to use "Fan" here. But the predefined mode "none" might also work, and be better than whatever happens now if you select fan_only from HVAC mode then try to change the preset mode.

Reply:
Yes it is indeed more straightforward to have fan only under the preset_mode. But I'm missing a nice fan icon. Why is it not visible when I changed this dp to:
```
    - id: 6
      name: preset_mode
      type: boolean
      mapping:
        - dps_val: false
          constraint: work_mode
          conditions:
            - dps_val: "level1"
              value: Fan only
              icon: "mdi:fan"
            - dps_val: "level2"
              value: comfort
            - dps_val: "level3"
              value: boost
        - dps_val: true
          value: eco
```

![image](https://github.com/make-all/tuya-local/assets/5202136/e5686ed8-eb6d-4e30-a69e-bf166e079b26)

Reply:
I can't see any way for an integration to define new preset icons.  The `icon` directive in the config files is for the entity icon.

Reply:
That's a pity.